app/Events/VoucherHistoryEmailRequested.php

```diff
@@ -18,17 +18,23 @@ class VoucherHistoryEmailRequested
 
     public $user;
     public $trader;
+    public $vouchers;
+    public $date;
+    public $max_date;
     public $file;
 
     /**
      * Create a new event instance.
      *
      * @return void
      */
-    public function __construct(User $user, Trader $trader, $file)
+    public function __construct(User $user, Trader $trader, $vouchers, $file, $date, $max_date = null)
```

Comment:
I thought about that too, but not sure it helps. Is the User just the logged in user? Can you get the vouchers from the trader? Could you get the dates inside the listener instead? Those sort of thoughts... but leaving as is 'works' so up to you/ time.